python/ray/tune/result_grid.py

```diff
@@ -15,30 +15,69 @@
 
 @PublicAPI(stability="beta")
 class ResultGrid:
-    """A set of ``Result`` objects returned from a call to ``tuner.fit()``.
+    """A set of ``Result`` objects for interacting with Ray Tune results.
 
     You can use it to inspect the trials run as well as obtaining the best result.
 
-    The constructor is a private API.
+    The constructor is a private API. This object can only be created as a result of
+    ``Tuner.fit()``.
 
     Usage pattern:
 
     .. code-block:: python
```

Comment:
can we test the docstrings here?

Reply:
Chatted with @krfricke offline. We can't actually test Ray code snippets in docstrings so we would have to skip everything anyways.

Also, Tune outputs won't really show up well when rendering the docstring.

Seems the best option right now is to leave as a python code block.